### PR TITLE
Redesigning/Localizing squareBracket

### PR DESCRIPTION
**Pull request by forkiesassds**
remake of #23 because @chaziz94 decided to do the funny and add useless crap to the feature branch

### CODE REVIEW

**Comment by chaziz94** [comment]
do we merge it on main or on redesign

it's made for the redesign in mind (even if we're testing translation on the main branch's ui aka prototype redesign)

**Comment by rollerozxa** [comment]
Wasn't the redesign unfinished?

**Comment by chaziz94** [comment]
yeah it is unfinished, but it was still made for the redesign.

image

there's a deprecated string due to the finobe-ish jumbotron getting removed in redesign. so not really worth translating

**Comment by rollerozxa** [comment]
Yeah we should probably merge i18n into redesign...

**Comment by chaziz94** [comment]
and great, Conflict.

**Comment by rollerozxa** [comment]
I honestly expected that. Shouldn't be too hard to go through and resolve manually...

**Comment by rollerozxa** [comment]
I went through and resolved every conflict. _layout.twig was reverted to the redesign, and the "What is squareBracket" section was reverted to the i18n branch which would need to be fixed manually as well.

**Comment by rollerozxa** [comment]
...Aand we're back on track, everything should be successfully merged and added back now. I'm fucking exhausted, I should probably go to bed before I totally collapse.

**Comment by chaziz94** [comment]
at this point the translation system is mostly complete other than like some weird thing with counting in russian that vc told me?

**Comment by forkiesassds** [comment]
oh yea, dates and relative time should also be localized.

File: templates/user.twig [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 	{% block content %}
 <div class="row" style="margin-top:-150px;">
-<p class="text-white">are you fucking kidding me guys this is literally just the poktube semantic ui profile layout but without the user infomation detail bullshit</p>
+<p class="text-white">{{ __("are you fucking kidding me guys this is literally just the poktube semantic ui profile layout but without the user infomation detail bullshit") }}</p>
```
Comment (forkiesassds) [inline templates/user.twig:16]: why do we need to localize placeholder text?

Comment (chaziz94) [reply templates/user.twig:16]: why not

Comment (forkiesassds) [reply templates/user.twig:16]: PLACEHOLDER TEXT IS PLACEHOLDER TEXT, NO ONE IS GOING  TO SEE IT IN THE FINAL PRODUCT  AND THUS DOES NOT  NEED TO BE LOCALIZED

**Comment by chaziz94** [comment]
is it even ready? the way views are counted in Russian isn't exactly perfect

**Comment by rollerozxa** [comment]
Well yeah that's true. I was mainly just seeing whether it could currently be merged into main without any conflicts, because you're not really able to do that on draft PRs.

Anyways, some things that would be needed to be done on the i18n side would be:
- [ ] Adding a way to change the language in the settings (let's do it from a hardcoded array language code to language name, that would probably be the easiest)
- [ ] Adding language support for relative time for
    - [ ] Candian French
    - [ ] Russian
- [ ] Fixing exact dates to use localized month names (Example for Swedish would be 'Maj' instead of 'May'). [This might be the way to go regarding that.](https://twig.symfony.com/doc/3.x/filters/format_date.html)

**Comment by chaziz94** [comment]
the video tag implementation should be put to redesign branch

also what's the mysql command for the tags

**Comment by chaziz94** [comment]
for any confusion: the fact that most of the worked intended for the redesign has been done on the localization branch, i've decided to just merge them both.

you could tell me this is useless, but the only commit that i18n didn't had was some width change (https://github.com/chazizsquarebracket/squarebracket/pull/20/commits/fdbdb80a225e99cf9fe06d145eeb066394fc30ab)

**Comment by chaziz94** [comment]
tried fixing gitignore but gboard kept screwing up

**Comment by rollerozxa** [comment]
I think I managed to fix it

**Comment by chaziz94** [comment]
merging it to master because everyone's doing their changes into i18n

**Pull request merged**